### PR TITLE
Timur katimullin tasks 71 80

### PR DESCRIPTION
**Pull request by linkouth**
Adds solutions 71-80 tasks.

### CODE REVIEW

File: tasks/71_binary-encode/binary-encode.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * of the passed binary string
+ * @example
+ * // return '01100110 01101001 01110010 01100101'
+ * binaryEncode('fire')
+ * @param {String} str
+ * @return {String}
+ */
+export function binaryEncode(str) {
+  return str === undefined ? '' :
+  str
```
Comment (vovcyan) [inline tasks/71_binary-encode/binary-encode.js]: Please format it

File: tasks/72_carrying/carrying.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * expects one argument and returns the sum
+ * @example
+ * // return 5
+ * addTogether(2, 3);
+ * // return 5
+ * addTogether(2)(3);
+ * @return {*}
+ */
+export function addTogether(...args) {
+  const firstArg = args[0];
```
Comment (vovcyan) [inline tasks/72_carrying/carrying.js]: ES6

File: tasks/73_check-for-palindromes/check-for-palindromes.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * @example
+ * // return true
+ * palindrome(("race car");
+ * @param {String} str
+ * @return {Boolean}
+ */
+export function palindrome(str) {
+  const filteredString = str.replace(/[\W, \s, _]/g, '').toLowerCase();
+  for (let i = 0; i < filteredString.length / 2; i++) {
+    if (filteredString[i] !== filteredString[filteredString.length - i - 1]) {
```
Comment (vovcyan) [inline tasks/73_check-for-palindromes/check-for-palindromes.js]: Split it

File: tasks/74_dna-pairing/dna-pairing.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * Take each character, get its pair, and return the results as
+ * a 2d array
+ * @example
+ * // return [["A","T"],["T","A"],["C","G"],["G","C"],["A","T"]]
+ * pairElement("ATCGA");
+ * @param {String} str
+ * @return {Array<Array<String>>}
+ */
+export function pairElement(str) {
+  let elements = str.split('');
```
Comment (vovcyan) [inline tasks/74_dna-pairing/dna-pairing.js]: const

File: tasks/76_deep-object-clone/deep-object-clone.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * var original = {a: 1, b:{c:10}};
+ * var copy = deepClone(original);
+ * @param {Object} obj
+ * @return {Object}
+ */
+export function deepClone(obj) {
+  if (typeof obj !== 'object') {
+    return obj;
+  }
+  let result = {};
```
Comment (vovcyan) [inline tasks/76_deep-object-clone/deep-object-clone.js]: const

File: tasks/78_valid-braces/valid-braces.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * @example
+ * // return true
+ * validBraces('([{}])');
+ * // return false
+ * validBraces('[({})](]');
+ * @param {String} str
+ * @return {Boolean}
+ */
+export function validBraces(str) {
+  let stack = [];
```
Comment (vovcyan) [inline tasks/78_valid-braces/valid-braces.js]: const

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/78_valid-braces/valid-braces.js (hunk 6 above)
Comment (linkouth) [reply tasks/78_valid-braces/valid-braces.js]: Fixed

File: tasks/76_deep-object-clone/deep-object-clone.js (hunk 5 above)
Comment (linkouth) [reply tasks/76_deep-object-clone/deep-object-clone.js]: Fixed

File: tasks/74_dna-pairing/dna-pairing.js (hunk 4 above)
Comment (linkouth) [reply tasks/74_dna-pairing/dna-pairing.js]: Fixed

File: tasks/73_check-for-palindromes/check-for-palindromes.js (hunk 3 above)
Comment (linkouth) [reply tasks/73_check-for-palindromes/check-for-palindromes.js]: Fixed

File: tasks/72_carrying/carrying.js (hunk 2 above)
Comment (linkouth) [reply tasks/72_carrying/carrying.js]: Fixed

File: tasks/71_binary-encode/binary-encode.js (hunk 1 above)
Comment (linkouth) [reply tasks/71_binary-encode/binary-encode.js]: Fixed

**Review by alexkond — APPROVED** [review]

**Review by vovcyan — APPROVED** [review]

**Pull request merged**